services/repository/mongodb/service.go

```diff
@@ -0,0 +1,173 @@
+// Package mongodb implements MongoDB repository services
+package mongodb
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/decentralized-cloud/tenant/models"
+	"github.com/decentralized-cloud/tenant/services/configuration"
+	"github.com/decentralized-cloud/tenant/services/repository"
+	commonErrors "github.com/micro-business/go-core/system/errors"
+	"go.mongodb.org/mongo-driver/bson"
+    "go.mongodb.org/mongo-driver/mongo"
+)
+
+type mongodbRepositoryService struct {
+	dbConnectionString string
+}
+
+// NewMongodbRepositoryService creates new instance of the RepositoryService, setting up all dependencies and returns the instance
+// Returns the new service or error if something goes wrong
+func NewMongodbRepositoryService(configurationService configuration.ConfigurationContract) (repository.RepositoryContract, error) {
+
+	if configurationService == nil {
+		return nil, commonErrors.NewArgumentNilError("configurationService", "configurationService is required")
+	}
+
+	dbConnectionString, err := configurationService.GetDbConnectionString()
+	if err != nil {
+		return nil, err
```

Comment:
You need to convert this error to repository errors defined in the error package